pkg/skaffold/build/cluster/sources/sources.go

```diff
@@ -77,6 +77,12 @@ func podTemplate(clusterDetails *latest.ClusterDetails, image string, args []str
 					},
 					Resources: resourceRequirements(clusterDetails.Resources),
 				},
+				{
+                    Name:            "logger-container",
+                    Image:           constants.DefaultBusyboxImage,
+                    ImagePullPolicy: v1.PullIfNotPresent,
+                    Command: []string{"sh", "-c", "while [[ $(ps -ef | grep kaniko | wc -l) -gt 1 ]] ; do   sleep 1; done; sleep " + clusterDetails.PodGracePeriodSeconds},
```

Comment:
hmm i am a little bit concerned with adding a sleep here.  I think the problem you are trying to solve here is when 
- kaniko pod dies and gets cleaned up by kube admin before we could steam logs?

is that correct?
Adding a grace time would not still able to fix this issue what if still dont end up reading them within the grace period.
I think the right approach would be kaniko should write logs to a file to some volume mounted path and we attach a side car container to read the logs and writes. https://kubernetes.io/docs/concepts/cluster-administration/logging/#using-a-sidecar-container-with-the-logging-agent

But again, thats a big change. Not sure what is the correct thing here.

Reply:
> kaniko pod dies and gets cleaned up by kube admin before we could steam logs? 

Exactly


> Adding a grace time would not still able to fix this issue what if still dont end up reading them within the grace period.

User can simply increase the grace period which is user configurable.

> I think the right approach would be kaniko should write logs to a file to some volume mounted path and we attach a side car container to read the logs and writes. 

Yes that could be a big change and is one of the possible solution. We can also make changes in kaniko for storing logs in some volume. Or maybe something even better.